client/webserver/api.go

```diff
@@ -316,6 +316,29 @@ func (s *WebServer) apiInit(w http.ResponseWriter, r *http.Request) {
 	s.actuallyLogin(w, r, login)
 }
 
+// apiShutdown is the handler for the `/shutdown` API request.
+func (s *WebServer) apiShutdown(w http.ResponseWriter, r *http.Request) {
+	shutdown := new(shutdownForm)
+	if !readPost(w, r, shutdown) {
+		return
+	}
+
+	// Force flag can be used to skip the logout call which might return an error
+	// in case any active orders found.
+	if !shutdown.Force {
+		err := s.core.Logout()
+		if err != nil {
+			s.writeAPIError(w, "shutdown error: %v", err)
+			return
+		}
+	}
+
+	// Cancel Core's context.
+	s.coreCancel()
+
+	writeJSON(w, simpleAck(), s.indent)
```

Comment:
Not sure, but I guess we are waiting for all handlers to finish after shutdown? If not would need to acknowledge first, but probably ok.